inspirehep/utils/record_getter.py

```diff
@@ -105,3 +106,16 @@ def get_db_record(pid_type, recid):
     from inspirehep.modules.records.api import InspireRecord
     pid = PersistentIdentifier.get(pid_type, recid)
     return InspireRecord.get_record(pid.object_uuid)
+
+
+def get_db_records(pid_type, recids):
+    """Get an iterator on record metadata from the DB."""
+    recids = [str(recid) for recid in recids]
+
+    query = RecordMetadata.query.join(
+        PersistentIdentifier, RecordMetadata.id == PersistentIdentifier.object_uuid
+    ).filter(
+        PersistentIdentifier.pid_value.in_(recids), PersistentIdentifier.pid_type == pid_type
+    )
+
+    return (record.json for record in query.all())
```

Comment:
should I rather use [ `yield_per`](http://docs.sqlalchemy.org/en/latest/orm/query.html#sqlalchemy.orm.query.Query.yield_per) here? the disclaimers are a bit scary, and I don't really understand what they mean.

Reply:
from that reply it looks safe, as we should have exactly one record associated to a given recid.

Reply:
Right, that objection doesn't apply here.